compiler/rustc_resolve/src/late.rs

```diff
@@ -3207,7 +3207,7 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
 
         // 3) Report all missing variables we found.
         let mut missing_vars = missing_vars.into_iter().collect::<Vec<_>>();
-        missing_vars.sort_by_key(|&(sym, ref _err)| sym);
+        missing_vars.sort_by(|(sym_a, _), (sym_b, _)| sym_a.as_str().cmp(sym_b.as_str()));
```

Comment:
Should `missing_vars` be a `FxIndexMap` from the beginning, to have them in source order and avoid any sorting?

Reply:
Likewise for `inconsistent_vars`?

Reply:
makes sense